gitter.py

```diff
@@ -216,6 +224,40 @@ def __unicode__(self):
 
     __str__ = __unicode__
 
+class GitterRoomThread(threading.Thread):
+    def __init__(self, room, backend):
+        super().__init__()
+        self.room = room
+        self.backend = backend
+
+    def run(self):
+        self.room.joined = True
+        log.debug("thread for %s started" % self.room.idd)
+        while True:
+            self.stream()
```

Comment:
Don't think that's enough/applicable here.

Also, this needs incremental backoff I think. With the current loop, if `streamAPIRequest` fails immediately, we would end up hammering Gitter with multiple requests/s.